tests/ecmascript/test-bi-reflect-apply-construct.js

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Reflect.apply()
+ *  Reflect.construct()
+ */
+
+/*===
+Casper the friendly ghost
+maggie the eaty pig
+Darkling the undefined
+===*/
+
+function applyTest() {
+    var func = function(x, y) {
+        if (arguments.length >= 2) {
+            print(this.name, "the", x, y);
+        } else {
+            print(this.name, "the", x);
+        }
+    }
+    var casper = { name: "Casper" };
+    var maggie = { name: "maggie" };
+    var darkling = { name: "Darkling" };
+    Reflect.apply(func, casper, [ "friendly", "ghost" ]);
+    Reflect.apply(func, maggie, [ "eaty", "pig" ]);
+    Reflect.apply(func, darkling, []);
+}
+
+try {
+    applyTest();
```

Comment:
Good point, I'll add something to exercise that.  I was honestly not sure what to do with the apply and construct tests; once I tested the basic functionality there didn't seem to be too many corner cases to worry about.